src/Raven.Server/Rachis/Follower.cs

```diff
@@ -467,6 +467,33 @@ private void NegotiateWithLeader(TransactionOperationContext context, LogLengthN
 
                 context.Transaction.Commit();
             }
+
+            _engine.Timeout.Defer(_connection.Source);
+            _debugRecorder.Record("Invoking StateMachine.SnapshotInstalled");
+
+            // notify the state machine, we do this in an async manner, and start
+            // the operator in a separate thread to avoid timeouts while this is
+            // going on
+
+            var task = Task.Run(() => _engine.SnapshotInstalledAsync(snapshot.LastIncludedIndex));
```

Comment:
Do we need to pass a cancelation token here, otherwise if we dispose of this Follower we can end up with a rogue task, no?

Reply:
We can't really get that, because the code will wait until the task is completed.
But that might not be a bad idea, yes.

Reply:
Just checked it, and there doesn't seem to be anything there that uses a cancellation token that we can pass, so don't think it will work